message_ix/tests/test_tutorials.py

```diff
@@ -1,85 +1,109 @@
     # NB could also check objective function values in the following tutorials; however,
     #    better to test features directly (not via Jupyter/IPython)
-    (("westeros", "westeros_baseline_using_xlsx_import_part1"), [], {}),
-    (("westeros", "westeros_baseline_using_xlsx_import_part2"), [], {}),
-    (("westeros", "westeros_emissions_bounds"), [], {}),
-    (("westeros", "westeros_emissions_taxes"), [], {}),
-    (("westeros", "westeros_firm_capacity"), [], {}),
-    (("westeros", "westeros_flexible_generation"), [], {}),
-    (("westeros", "westeros_fossil_resource"), [], {}),
-    (("westeros", "westeros_share_constraint"), [], {}),
-    (("westeros", "westeros_soft_constraints"), [], {}),
-    (("westeros", "westeros_addon_technologies"), [], {}),
-    (("westeros", "westeros_historical_new_capacity"), [], {}),
-    (("westeros", "westeros_multinode"), [], {}),
+    _t("w0", f"{W}_baseline_using_xlsx_import_part1"),
+    _t("w0", f"{W}_baseline_using_xlsx_import_part2"),
+    _t("w0", f"{W}_emissions_bounds"),
+    _t("w0", f"{W}_emissions_taxes"),
+    _t("w0", f"{W}_firm_capacity"),
+    _t("w0", f"{W}_flexible_generation"),
+    _t("w0", f"{W}_fossil_resource"),
+    _t("w0", f"{W}_share_constraint"),
+    _t("w0", f"{W}_soft_constraints"),
+    _t("w0", f"{W}_addon_technologies"),
+    _t("w0", f"{W}_historical_new_capacity"),
+    _t("w0", f"{W}_multinode"),
     # NB this is the same value as in test_reporter()
-    (("westeros", "westeros_report"), [("len-rep-graph", 13074)], {}),
-    ((AT, "austria"), [("solve-objective-value", 206321.90625)], {}),
-    (
-        (AT, "austria_single_policy"),
-        [("solve-objective-value", 205310.34375)],
-        {},
-    ),
-    ((AT, "austria_multiple_policies"), [], {}),
-    ((AT, "austria_multiple_policies-answers"), [], {}),
-    ((AT, "austria_load_scenario"), [], {}),
+    _t(None, f"{W}_report", check=[("len-rep-graph", 13074)], marks=[MARK[2]]),
+    _t("at0", "austria", check=[("solve-objective-value", 206321.90625)]),
+    _t("at0", "austria_single_policy", check=[("solve-objective-value", 205310.34375)]),
+    _t("at0", "austria_multiple_policies"),
+    _t("at0", "austria_multiple_policies-answers"),
+    _t("at0", "austria_load_scenario"),
     # R tutorials using the IR Jupyter kernel
-    pytest.param((AT, "R_austria"), [], dict(kernel="IR"), marks=[mark0, mark1, mark2]),
-    pytest.param(
-        (AT, "R_austria_load_scenario"), [], dict(kernel="IR"), marks=[mark0, mark2]
-    ),
+    _t("at1", "R_austria", marks=MARK[:3]),
+    _t("at1", "R_austria_load_scenario", marks=[MARK[0], MARK[2]]),
 ]
 
 # Short, readable IDs for the tests. Use getattr() to unpack the values from
 # pytest.param()
-ids = [getattr(arg, "values", arg)[0][-1] for arg in tutorials]
+IDS = [getattr(arg, "values", arg)[0][-1] for arg in TUTORIALS]
 
-# List of data files required to run tutorials
-data_files = [
+#: List of data files required by "westeros_baseline_using_xlsx_import_part2".
```

Comment:
What is the purpose of this `:`?

Reply:
These are picked up by sphinx.ext.autodoc. See [here](https://www.sphinx-doc.org/en/master/usage/extensions/autodoc.html#directive-automethod) in the para beginning "For module data members and class attributes, …"

Reply:
Thanks :)